src/DSharp.Mscorlib/Collections/Generic/List.cs

```diff
@@ -10,127 +11,70 @@ namespace System.Collections.Generic
     [ScriptIgnoreNamespace]
     [ScriptImport]
     [ScriptName("Array")]
-    public sealed class List<T> : ICollection<T>
+    public sealed partial class List<T> : IList<T>, IList, IReadOnlyList<T>
     {
-        public List()
-        {
-        }
+        public List() { }
 
-        public List(int capacity)
-        {
-        }
+        public List(int capacity) { }
 
-        public List(params T[] items)
-        {
-        }
+        public List(params T[] items) { } /*items type should be IEnumerable<T> and not params T*/
 
         [ScriptField]
         [ScriptName("length")]
-        public int Count { get; }
+        public extern int Count { get; }
 
         [ScriptField]
-        public extern T this[int index] { get; set; }
+        extern T IReadOnlyList<T>.this[int index] { get; }
+
+        [ScriptField]
+        extern T IList<T>.this[int index] { get; set; }
+
+        [ScriptField]
+        public extern object this[int index] { get; set; }
 
         [ScriptName("push")]
         public extern void Add(T item);
 
         [ScriptName("push")]
-        public extern void AddRange(params T[] items);
-
-        public extern void Clear();
+        public extern int Add(object value);
 
-        public extern List<T> Concat(params T[] objects);
+        public extern bool Contains(T item);
 
-        public extern bool Contains(object item);
-
-        public extern bool Every(ListFilterCallback<T> filterCallback);
-
-        public extern bool Every(ListItemFilterCallback<T> itemFilterCallback);
-
-        public extern List<T> Filter(ListFilterCallback<T> filterCallback);
-
-        public extern List<T> Filter(ListItemFilterCallback<T> itemFilterCallback);
-
-        public extern void ForEach(ListCallback<T> callback);
+        public extern bool Contains(object value);
 
-        public extern void ForEach(ListItemCallback<T> itemCallback);
-
-        public extern IEnumerator<T> GetEnumerator();
-
-        extern IEnumerator IEnumerable.GetEnumerator();
-
-        public extern List<T> GetRange(int index);
+        public extern void Clear();
 
-        public extern List<T> GetRange(int index, int count);
+        public extern int IndexOf(object value);
 
         public extern int IndexOf(T item);
 
-        public extern int IndexOf(T item, int startIndex);
-
-        public extern void Insert(int index, T item);
-
-        public extern void InsertRange(int index, params T[] items);
-
-        public extern string Join();
-
-        public extern string Join(string delimiter);
-
-        public extern int LastIndexOf(object item);
+        //In CLR we also have this signature -> IndexOf(T item, int index, int count)
```

Comment:
Misplaced comment?

Reply:
It was a comment pointing out that we dont support one of the overloads, im fine with keeping it like that